src/components/postsSlice/postsSlice.ts

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable no-param-reassign */
+import { PayloadAction, createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { getUserPosts } from '../../api/posts';
+import { Post } from '../../types/Post';
+
+export const fetchPosts = createAsyncThunk(
+  'posts/fetchUserPosts',
+  async (userId: number) => {
+    const userPosts = await getUserPosts(userId);
+
+    return userPosts;
+  },
+);
+
+type PostsState = {
+  items: Post[];
+  selectedPost: Post | null;
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: PostsState = {
+  items: [],
+  selectedPost: null,
+  loaded: false,
```

Comment:
You can use as a template other slices. 
Try to use similar main constants ( loading, error, etc.) 
In each slice, we need to have a base constant that response to loading, error, and elements. Then you can add others for specific cases.

Reply:
I'm not sure if I understood correctly about template) I changed names only.